nominal/core/_clientsbunch.py

```diff
@@ -64,3 +69,39 @@ def from_config(cls, cfg: ServiceConfiguration, agent: str, token: str) -> Self:
             upload=client_factory(upload_api.UploadService),
             video=client_factory(scout_video.VideoService),
         )
+
+    def auth_expiration(self) -> datetime.datetime:
+        """Returns the timestamp at which the current auth token becomes invalid
+
+        Raises:
+            ValueError: Malformed auth token present
+        """
+        token_parts = self.auth_header.split()
+        if len(token_parts) != 2:
+            raise ValueError(f"Could not parse auth header: expected two space-separated parts")
+
+        token = token_parts[1]
+        payload = jwt.decode(token, options={"verify_signature": False}, algorithms=["RS256"])
+        expiry_s = payload.get("exp")
+        if expiry_s is None:
+            raise ValueError("Expected expiry to be present in auth token, but not present")
+
+        return datetime.datetime.fromtimestamp(expiry_s, datetime.timezone.utc)
+
+    def auth_valid(self) -> bool:
+        """Validates that authentication tokens are valid and we can successfully hit the nominal API."""
+        try:
+            # simplest possible API call with minimal overhead to validate conectivity and
+            # proper authentication
+            self.authentication.get_my_profile(self.auth_header)
+            return True
```

Comment:
this already exists on the client via `get_user()`: https://github.com/nominal-io/nominal-client/blob/ff6f015bfba7a95617985763df1df4871eb8baee/nominal/core/client.py#L73-L75